pkg/client-sdk/covenantless_client.go

```diff
@@ -2702,3 +2771,33 @@ func buildRedeemTx(
 
 	return bitcointree.BuildRedeemTx(ins, outs)
 }
+
+func handleOptions(options *SettleOptions) ([]*secp256k1.PrivateKey, []string, tree.SigningType, error) {
+	signerPrivKeys := make([]*secp256k1.PrivateKey, 0)
+	signerPubKeys := make([]string, 0)
+	if len(options.CustomSignerPrivKeys) > 0 {
+		for _, privKey := range options.CustomSignerPrivKeys {
+			signerPrivKeys = append(signerPrivKeys, privKey)
+			signerPubKeys = append(signerPubKeys, hex.EncodeToString(privKey.PubKey().SerializeCompressed()))
+		}
+	}
+
+	var signingType tree.SigningType
+	if options.SigningType != nil {
+		signingType = *options.SigningType
+	} else {
+		signingType = tree.SignBranch
+	}
+
+	// if no custom signer priv keys are provided, we generate a new ephemeral key
+	if len(signerPubKeys) == 0 {
```

Comment:
in light of [this comment](https://github.com/ark-network/ark/pull/422/files#r1918586259), we should always generate the keypair for the sdk wallet